src/lib.rs

```diff
@@ -170,6 +173,22 @@ impl Client {
         self
     }
 
+    /// Used to specify a header that will be passed to all queries.
+    ///
+    /// # Example
+    /// ```
+    /// # use clickhouse::Client;
+    /// Client::default().with_header(http::header::COOKIE, http::HeaderValue::from_static("A=1"));
+    /// ```
+    pub fn with_header(
+        mut self,
+        name: impl Into<HeaderName>,
```

Comment:
I think we can accept `impl TryInto` to make it possible to pass strings. Downside here is that we cannot use `HeaderMap` and should use `HashMap<Result<_>, Result<_>>` instead =(